ui/hooks/useTransactionEventFragment.js

```diff
@@ -23,13 +20,12 @@ export const useTransactionEventFragment = () => {
       if (!transaction || !transaction.id) {
         return;
       }
-      if (!fragment) {
-        await createTransactionEventFragment(
-          transaction.id,
-          TransactionMetaMetricsEvent.approved,
-        );
-      }
-      updateEventFragment(`transaction-added-${transaction.id}`, params);
+      await createOrUpdateTransactionEventFragment(
```

Comment:
Does the UI layer have to change to support this?

We can still provide the same action and underlying method exactly as it was, it's just that the supporting logic is no longer inside the `TransactionController`.

It would shrink this PR considerably and minimise some of the risk too.